usr/src/uts/common/fs/zfs/arc.c

```diff
@@ -2931,9 +2943,34 @@ arc_evict_hdr(arc_buf_hdr_t *hdr, kmutex_t *hash_lock)
 	return (bytes_evicted);
 }
 
+/*
+ * Based on l2arc_spa_list, returns true if the
+ * given spa has an alive (!dead) L2 device,
+ * false otherwise.
+ */
+static boolean_t
+l2arc_alive(uint64_t spa)
```

Comment:
Could you do this by walking the existing `l2arc_dev_list`?  This mechanism seems quite complicated.

That said, I'd be concerned about the performance of this either way, it's in a pretty hot path.